src/views/text_input.rs

```diff
@@ -947,426 +627,221 @@ impl Widget for TextInput {
-                cx.app_state_mut().request_layout(self.id());
-                if cx.is_active(self.id()) {
-                    let selection_stop = self.get_box_position(event.pos.x, event.pos.y, cx);
-                    self.update_selection(self.cursor_glyph_idx, selection_stop);
+                TextInputState::ContentExternal(text) => {
+                    self.content = text;
+                    self.text_layout.set(None);
+                }
+                TextInputState::Focus(focus) => {
+                    self.focused = focus;
+                    // NOTE: Moving cursor for pointer focus is handled in the event method, since we need to know
+                    // the click pos
+                    if !self.pointer_focus.get_untracked() {
+                        self.cursor.update(|cursor| {
+                            cursor.set_insert(Selection::caret(self.content.len()));
+                        });
+                    }
+                }
+                TextInputState::DocumentUpdated => {
+                    if let Some(func) = self.on_update_fn.as_ref() {
+                        func(self.content.clone())
+                    }
+                }
+                TextInputState::Placeholder(placeholder) => {
+                    self.placeholder = placeholder;
+                    self.placeholder_text_layout = None;
                 }
-                false
             }
-            Event::KeyDown(event) => self.handle_key_down(cx, event),
-            _ => false,
-        };
-
-        if is_handled {
-            cx.app_state.request_layout(self.id());
-            self.last_cursor_action_on = Instant::now();
+            cx.request_layout(self.id);
         }
-
-        EventPropagation::Continue
     }
 
     fn style(&mut self, cx: &mut crate::context::StyleCx<'_>) {
         let style = cx.style();
-        if self.font.read(cx) || self.text_buf.is_none() {
-            self.update_text_layout();
-            cx.app_state_mut().request_layout(self.id());
+        let id = self.id();
+        let placeholder_style = style.clone().apply_class(PlaceholderTextClass);
+        if self.placeholder_style.read_style(cx, &placeholder_style) {
+            cx.app_state_mut().request_paint(id);
         }
+
         if self.style.read(cx) {
-            cx.app_state_mut().request_paint(self.id());
+            self.set_text_layout();
+            cx.app_state_mut().request_layout(id);
         }
 
-        self.selection_style.read_style(cx, &style);
-
-        let placeholder_style = style.clone().apply_class(PlaceholderTextClass);
-        self.placeholder_style.read_style(cx, &placeholder_style);
+        // To set wrap method to `None`
+        self.editor.with_untracked(|ed| {
+            ed.es.update(|s| {
+                if s.read(cx) {
+                    ed.floem_style_id.update(|val| *val += 1);
+                    cx.app_state_mut().request_paint(id);
+                }
+            })
+        });
```

Comment:
I'm confused what this has to do with setting wrap to `None`

Reply:
Well usually the *text_editor* view reads  EditorStyle, but since I am not using it, I need to do the same in the input to get it